keras_nlp/models/preprocessor.py

```diff
@@ -0,0 +1,127 @@
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+
+    @property
+    def tokenizer(self):
+        """The tokenizer used to tokenize strings."""
+        return self._tokenizer
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "tokenizer": keras.layers.serialize(self.tokenizer),
+                "sequence_length": self.packer.sequence_length,
+                "truncate": self.packer.truncate,
+            }
+        )
+        return config
+
+    @classmethod
+    def from_config(cls, config):
+        if "tokenizer" in config and isinstance(config["tokenizer"], dict):
+            config["tokenizer"] = keras.layers.deserialize(config["tokenizer"])
+        return cls(**config)
+
+    @classproperty
+    def tokenizer_cls(cls):
+        return None
+
+    @classproperty
+    def presets(cls):
+        return {}
+
+    @classmethod
+    def from_preset(
+        cls,
+        preset,
+        sequence_length=None,
+        **kwargs,
+    ):
+        """Instantiate {{model_name}} preprocessor from preset architecture.
+
+        Args:
+            preset: string. Must be one of "{{preset_names}}".
+            sequence_length: int, optional. The length of the packed inputs.
+                Must be equal to or smaller than the `max_sequence_length` of
+                the preset. If left as default, the `max_sequence_length` of
+                the preset will be used.
+
+        Examples:
+        ```python
+        # Load preprocessor from preset
+        preprocessor = keras_nlp.models.{{model_name}}Preprocessor.from_preset(
+            "{{example_preset_name}}",
+        )
+        preprocessor("The quick brown fox jumped.")
+
+        # Override sequence_length
+        preprocessor = keras_nlp.models.{{model_name}}Preprocessor.from_preset(
+            "{{example_preset_name}}",
+            sequence_length=64
+        )
+        preprocessor("The quick brown fox jumped.")
+        ```
+        """
+        if not cls.presets:
+            raise NotImplementedError(
+                "No presets have been created for this class."
+            )
+        if preset not in cls.presets:
+            raise ValueError(
+                "`preset` must be one of "
+                f"""{", ".join(cls.presets)}. Received: {preset}."""
+            )
+
+        tokenizer = cls.tokenizer_cls.from_preset(preset)
+
+        metadata = cls.presets[preset]
+        # For task model presets, the backbone config is nested.
+        if "backbone" in metadata["config"]:
+            backbone_config = metadata["config"]["backbone"]["config"]
+        else:
+            backbone_config = metadata["config"]
+
+        # Use model's `max_sequence_length` if `sequence_length` unspecified;
+        # otherwise check that `sequence_length` not too long.
+        max_sequence_length = backbone_config["max_sequence_length"]
+        if sequence_length is not None:
```

Comment:
if we switch to routing everything through kwargs, we can just do this check via `kwargs["sequence_length"]`

Reply:
Yeah, we can do that. I was following this comment: https://github.com/keras-team/keras-nlp/issues/636#issue-1520042088. Let me know which way you want it!